external/llvm-project/mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2805,15 +2805,31 @@ Value mlir::LLVM::createGlobalString(Location loc, OpBuilder &builder,
   auto module =
       builder.getInsertionBlock()->getParentOp()->getParentOfType<ModuleOp>();
   assert(module && "builder points to an op outside of a module");
-
-  // Create the global at the entry of the module.
-  OpBuilder moduleBuilder(module.getBodyRegion(), builder.getListener());
   MLIRContext *ctx = builder.getContext();
-  auto type = LLVM::LLVMArrayType::get(IntegerType::get(ctx, 8), value.size());
-  auto global = moduleBuilder.create<LLVM::GlobalOp>(
-      loc, type, /*isConstant=*/true, linkage, name,
-      builder.getStringAttr(value), /*alignment=*/0);
 
+  SmallString<64> uniqueName(name);
+
+  // Create the global at the entry of the module.
+  // Uniquify if necessary or return found equivalent.
+  LLVM::GlobalOp global = module.lookupSymbol<LLVM::GlobalOp>(uniqueName);
+  APInt i(32, 0);
```

Comment:
I'm pretty sure this can be an `uint32_t` and then we can use `Twine(i)`

Reply:
yeah, it is possible. I will probably follow it up in another PR if this passes CI.